arch/arm/core/fault.c

```diff
@@ -164,13 +165,34 @@ static void _FaultShow(const NANO_ESF *esf, int fault)
 /* HardFault is used for all fault conditions on ARMv6-M. */
 #elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
 
+#ifdef CONFIG_USERSPACE
+Z_EXC_DECLARE(z_arch_user_string_nlen);
+
+static const struct z_exc_handle exceptions[] = {
+	Z_EXC_HANDLE(z_arch_user_string_nlen)
+};
+#endif
+
 /* Perform an assessment whether an MPU fault shall be
  * treated as recoverable.
  *
  * @return 1 if error is recoverable, otherwise return 0.
  */
-static int _MpuFaultIsRecoverable(const NANO_ESF *esf)
+static int _MemoryFaultIsRecoverable(NANO_ESF *esf)
```

Comment:
I am fine with changing MPU to Memory, if this function is to be used in a more generic way.

Reply:
@andrewboie , in the  amended version of this PR, this function is also used in ARMv6-M, so it needs to be  implemented outside the #ifdef ARMV7-M guard.